contracts/contracts/GrantRegistry.sol

```diff
@@ -1,135 +1,124 @@
+    grantCount += 1;
   }
 
-  function isActive(bytes32 _id) public view returns (bool) {
-    return grants[_id].state == State.ACTIVE;
+  /**
+   * @notice Update the owner of a grant
+   * @param _id ID of grant to update
+   * @param _owner New owner address
+   */
+  function updateGrantOwner(uint96 _id, address _owner) external {
+    Grant storage grant = grants[_id];
+    require(msg.sender == grant.owner, "Not authorized");
+    grant.owner = _owner;
+    emit GrantUpdated(grant.id, grant.owner, grant.payee, grant.metaPtr);
   }
 
-  function isClosed(bytes32 _id) public view returns (bool) {
-    return grants[_id].state == State.CLOSED;
+  /**
+   * @notice Update the payee of a grant
+   * @param _id ID of grant to update
+   * @param _payee New payee address
+   */
+  function updateGrantPayee(uint96 _id, address _payee) external {
+    Grant storage grant = grants[_id];
+    require(msg.sender == grant.owner, "Not authorized");
+    grant.payee = _payee;
+    emit GrantUpdated(grant.id, grant.owner, grant.payee, grant.metaPtr);
   }
 
-  function getGrantCount() public view returns (uint256) {
-    return grantCount;
+  /**
+   * @notice Update the payee of a grant
+   * @param _id ID of grant to update
+   * @param _metaPtr New URL that points to grant metadata
+   */
+  function updateGrantMetaPtr(uint96 _id, string calldata _metaPtr) external {
+    Grant storage grant = grants[_id];
+    require(msg.sender == grant.owner, "Not authorized");
+    grant.metaPtr = _metaPtr;
+    emit GrantUpdated(grant.id, grant.owner, grant.payee, grant.metaPtr);
   }
 
-  function getAllGrantIds() public view returns (bytes32[] memory) {
-    bytes32[] memory ret = new bytes32[](grantCount);
-    for (uint256 i = 0; i < grantCount; i++) {
-      ret[i] = grantIds[i];
-    }
-    return grantIds;
+  /**
+   * @notice Update multiple fields of a grant at once
+   * @dev To leave a field unchanged, you must pass in the same value as the current value
+   * @param _id ID of grant to update
+   * @param _owner New owner address
+   * @param _payee New payee address
+   * @param _metaPtr New URL that points to grant metadata
+   */
+  function updateGrant(
+    uint96 _id,
+    address _owner,
+    address _payee,
+    string calldata _metaPtr
+  ) external {
+    Grant storage grant = grants[_id];
+    require(msg.sender == grant.owner, "Not authorized");
+    grant.owner = _owner;
+    grant.payee = _payee;
+    grant.metaPtr = _metaPtr;
+    emit GrantUpdated(grant.id, grant.owner, grant.payee, grant.metaPtr);
   }
 
+  // --- View functions ---
+  /**
+   * @notice Returns an array of all grants and their on-chain data
+   * @dev May run out of gas for large values `grantCount`, depending on the node's RpcGasLimit. In these cases,
+   * `getGrants` can be used to fetch a subset of grants and aggregate the results of various calls off-chain
+   */
   function getAllGrants() public view returns (Grant[] memory) {
```

Comment:
Can be `external` instead of `public`

Reply:
Fixed in 756f836a84c1f1b47bf58f12734df5e871fd6b50